spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/core/SendingOperation.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.integration.core;
+
+import java.util.concurrent.CompletableFuture;
+
+
+/**
+ * Operations for sending message to a destination.
+ *
+ * @param <D> the type of message
+ * @author Warren Zhu
+ */
+public interface SendingOperation<D> {
```

Comment:
Send or SendMessage?

Reply:
Do we know type of message at compile time? Wondering if a generic interface is needed here.

Reply:
In Java's convention, async method should be suffixed with Async. Generic type is used in `EventHubOperation`, we also have similar `ServiceBusOperation` which has different message type